### PR TITLE
Migrate tests

### PR DESCRIPTION
**Pull request by martincostello**
As part of #380 (and eventually #471), this PR migrates some of the "old style" integration tests to the "newer way" of doing our integration tests.

These tests still use the queue classes directly in an _integration-y_ way, but after this is merged I'll migrate them to setup via the fluent API so they're less "low-level".

The only "old" integration tests left now are the multi-region ones, which need a real AWS account to refactor them against as the emulator doesn't really support multi-region.

### CODE REVIEW

**Comment by AnthonySteele** [comment]
Hard to review, since there are a lots of files and they don't match up - name, folder and contents have all changed so git can't tell which new one matches which old one.

**Comment by martincostello** [comment]
All I can suggest is review one commit at a time - they're a rewrite, so they won't align, and the styles are fundamentally different anyway.

Alternatively, I can cherry pick each commit into a different PR each, but that's basically the same as looking at the diffs in each commit, except split over 6 PRs instead of one.

File: JustSaying.IntegrationTests/Fluent/AwsTools/WhenCreatingErrorQueue.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+{
+    public class WhenCreatingErrorQueue : IntegrationTestBase
+    {
+        public WhenCreatingErrorQueue(ITestOutputHelper outputHelper)
+            : base(outputHelper)
+        {
+        }
+
+        [AwsFact]
+        public async Task Then_The_Retention_Period_Stays_At_Maximum()
```
Comment (adammorr) [inline JustSaying.IntegrationTests/Fluent/AwsTools/WhenCreatingErrorQueue.cs]: This test name is a bit misleading as it looks like we're asserting on the retention change (14 days to 100 seconds)

Comment (martincostello) [reply JustSaying.IntegrationTests/Fluent/AwsTools/WhenCreatingErrorQueue.cs]: 39b923565c1d6e41d96b86e04b93150026669179

**Pull request merged**